applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.hpp

```diff
@@ -0,0 +1,204 @@
+    {
+        // Get the initial properties from the model part. Recall that we create a separate
+        // properties object with reduced c and phi for each and every element. Those reduced
+        // properties objects are not linked to the original ones.
+        const auto& part_properties = mrModelPart.GetProperties(rProp.Id());
+
+        if (!part_properties.Has(INDEX_OF_UMAT_C_PARAMETER) ||
+            !part_properties.Has(NUMBER_OF_UMAT_PARAMETERS) || !part_properties.Has(UMAT_PARAMETERS)) {
+            KRATOS_ERROR << "Insufficient material data for C-phi reduction: " << std::endl;
+        } else {
+            if (part_properties[INDEX_OF_UMAT_C_PARAMETER] < 1 ||
+                part_properties[INDEX_OF_UMAT_C_PARAMETER] > part_properties[NUMBER_OF_UMAT_PARAMETERS]) {
+                KRATOS_ERROR << "undefined INDEX_OF_UMAT_C_PARAMETER: "
+                             << part_properties[INDEX_OF_UMAT_C_PARAMETER] << std::endl;
+            }
+            const auto c = part_properties[UMAT_PARAMETERS][part_properties[INDEX_OF_UMAT_C_PARAMETER] - 1];
+            if (c < 0.) KRATOS_ERROR << "Cohesion C out of range: " << c << std::endl;
+            return c;
+        }
+    }
+
+    double GetAndCheckYoung(const Element::PropertiesType& rProp)
+    {
+        const auto young = mrModelPart.GetProperties(rProp.Id())[UMAT_PARAMETERS][0];
+        if (young < 0.)
+            KRATOS_ERROR << "Positive value expected for Youngs modulus UMAT_PARAMETERS(1) "
+                         << young << std::endl;
+        return young;
+    }
+
+    double GetAndCheckPoisson(const Element::PropertiesType& rProp)
+    {
+        const auto nu = mrModelPart.GetProperties(rProp.Id())[UMAT_PARAMETERS][1];
+        if (nu < -1. || nu >= 0.5)
+            KRATOS_ERROR
+                << "Value between -1.0 and 0.5 expected for Poissons ratio UMAT_PARAMETERS(2) "
+                << nu << std::endl;
+        return nu;
+    }
+
+    void set_C_Phi_At_Element(Element& rElement, double ReducedPhi, double ReducedC) const
+    {
+        // Get C/Phi material properties of this element
+        const auto& r_prop = rElement.GetProperties();
+
+        // Overwrite C and Phi in the UMAT_PARAMETERS
+        auto Umat_parameters                                     = r_prop[UMAT_PARAMETERS];
+        Umat_parameters[r_prop[INDEX_OF_UMAT_PHI_PARAMETER] - 1] = ReducedPhi;
+        Umat_parameters[r_prop[INDEX_OF_UMAT_C_PARAMETER] - 1]   = ReducedC;
+
+        // Write back to the element
+        SetValueAtElement(rElement, UMAT_PARAMETERS, Umat_parameters);
+    }
+
+    void SetValueAtElement(Element& rElement, const Variable<Vector>& rVariable, const Vector& rValue) const
+    {
+        // Copies properties
+        Properties::Pointer p_new_prop = Kratos::make_shared<Properties>(rElement.GetProperties());
+
+        // Adds new properties to the element
+        p_new_prop->SetValue(rVariable, rValue);
+        rElement.SetProperties(p_new_prop);
```

Comment:
I'm not sure if it's necessary to make a copy here, I think this would do the same:
```suggestion
        rElement.GetProperties().SetValue(rVariable, rValue);
```

Reply:
It is not clear why but this change affects the solution. I will continue checking.

Reply:
Perhaps, initially the properties of the element point to the properties of the model part? If you start to change those, you will overwrite the initial values of the cohesion and the friction angle. However, those initial values are needed to compute the reduced values. By making a copy, we ensure that the initial values are kept as-is. But please, do check this hypothesis.